### DISCUSSION TITLE
How to get router id (ip address)

### DISCUSSION THREAD

**Discussion by orangepineapple** [ANLAB-KAIST/KENSv3#251] [Project Q&A 3. Internet Layer (IP/Routing)]
To send the packets we need the router ID, which is the IP address of our current router. The parameter to pwospfquery(ipv4 ) seems to be the id of the router we want to find a path to. So how do we get our own IP address? If we use std::optional<ipv4_t> ip = getIPAddr(port); which port should we use?

**Comment by Pusnow** [comment]
> If we use std::optional<ipv4_t> ip = getIPAddr(port); which port should we use?

You should use port 0 as stated in the documentation.
